### PR TITLE
IGNITE-10031 REST: Added "excludeCaches" parameter for "top" command.

### PR DESCRIPTION
**Pull request by akuznetsov-gridgain**

### CODE REVIEW

File: modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/JettyRestProcessorAbstractSelfTest.java [hunk 1]
```diff
@@ -1428,6 +1428,28 @@ public void testTopology() throws Exception {
                 assertEquals(publicCache.getConfiguration(CacheConfiguration.class).getCacheMode(), cacheMode);
             }
         }
+
+        // Test that caches were excluded.
+        ret = content(null, GridRestCommand.TOPOLOGY,
```
Comment (nva) [inline modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/JettyRestProcessorAbstractSelfTest.java]: Please, add same test for NODE command

File: modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/JettyRestProcessorAbstractSelfTest.java [hunk 2]
```diff
@@ -1428,6 +1428,28 @@ public void testTopology() throws Exception {
...
+            "attr", "false",
+            "mtr", "false",
+            "excludeCaches", "true"
```
Comment (nva) [inline modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/JettyRestProcessorAbstractSelfTest.java]: Add test for behavior without excludeCaches parameter (to check backward compatibility).

File: modules/core/src/main/java/org/apache/ignite/internal/processors/rest/handlers/top/GridTopologyCommandHandler.java [hunk 3]
```diff
@@ -216,16 +223,18 @@ private GridClientNodeBean createNodeBean(ClusterNode node, boolean mtr, boolean
         nodeBean.setTcpAddresses(nonEmptyList(node.<Collection<String>>attribute(ATTR_REST_TCP_ADDRS)));
         nodeBean.setTcpHostNames(nonEmptyList(node.<Collection<String>>attribute(ATTR_REST_TCP_HOST_NAMES)));
 
-        Map<String, CacheConfiguration> nodeCaches = ctx.discovery().nodePublicCaches(node);
+        if (!excludeCaches) {
+            Map<String, CacheConfiguration> nodeCaches = ctx.discovery().nodePublicCaches(node);
 
-        Collection<GridClientCacheBean> caches = new ArrayList<>(nodeCaches.size());
+            ArrayList<GridClientCacheBean> caches = new ArrayList<>(nodeCaches.size());
```
Comment (nva) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/rest/handlers/top/GridTopologyCommandHandler.java]:
```suggestion
            Collection<GridClientCacheBean> caches = new ArrayList<>(nodeCaches.size());
```

File: modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/JettyRestProcessorAbstractSelfTest.java (hunk 1 above)
Comment (akuznetsov-gridgain) [reply modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/JettyRestProcessorAbstractSelfTest.java]: Added

File: modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/JettyRestProcessorAbstractSelfTest.java (hunk 2 above)
Comment (akuznetsov-gridgain) [reply modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/JettyRestProcessorAbstractSelfTest.java]: Actually it is tested by first part of this test.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/rest/handlers/top/GridTopologyCommandHandler.java (hunk 3 above)
Comment (akuznetsov-gridgain) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/rest/handlers/top/GridTopologyCommandHandler.java]: Fixed

**Pull request closed without merging**